### PR TITLE
DOC: add examples to insert and update generally

### PR DESCRIPTION
**Pull request by erictleung**
This commit mainly adds examples on usage. This commit also fills in
information suggested by `validate_docstrings.py` script.

Ran

```
python scripts/validate_docstrings.py pandas.DataFrame.insert
```

to check docs and added in suggested information (e.g., See Also section).

- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3677,6 +3677,8 @@ def insert(self, loc, column, value, allow_duplicates=False) -> None:
         """
         Insert column into DataFrame at specified location.
 
+        Performs column insertion in-place.
```
Comment (arw2019) [inline pandas/core/frame.py:3680]: I don't think this is necessary. Pretty much this is what the line above says.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -3687,7 +3689,31 @@ def insert(self, loc, column, value, allow_duplicates=False) -> None:
         column : str, number, or hashable object
             Label of the inserted column.
         value : int, Series, or array-like
+            Input data to be inserted.
```
Comment (arw2019) [inline pandas/core/frame.py:3692]: I think this is self-explanatory so I'd remove this description

File: pandas/core/frame.py [hunk 3]
```diff
@@ -3687,7 +3689,31 @@ def insert(self, loc, column, value, allow_duplicates=False) -> None:
...
         allow_duplicates : bool, optional
+            Whether to allow duplicate column label names.
```
Comment (arw2019) [inline pandas/core/frame.py:3694]: Same as above, I think it's all in the name

**Review by arw2019 — COMMENTED** [review]
Thanks @erictleung for the PR! 

Small comments, LGTM otherwise

**Comment by erictleung** [comment]
@arw2019 changes made. Thanks for the feedback!

**Review by arw2019 — APPROVED** [review]

**Comment by mroeschke** [comment]
Sorry this PR has been unaddressed for a while. LGTM but could you merge master one more time?

**Comment by erictleung** [comment]
@mcalingo thanks for taking a look at this! I rebased my branch onto the most recent commits on `master`. That should resolve any issues merging. Let me know if there are any issues going forward in merging.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @erictleung

**Pull request merged**